scripts/dts/devicetree.py

```diff
@@ -123,6 +124,9 @@ def parse_value(value):
             return int(value, 16)
         if value[0] == '0':
             return int(value, 8)
+        # Match alpha numeric values
+        if re.match("\w", value):
+            return value
```

Comment:
This parsing code needs a lot of cleanup in general I think. It doesn't make sense to re-enter `parse_value()` for the contents of a string.

Could you try to come up with some version where strings don't re-enter `parse_value()`? If I'm reading this right, strings starting with `"0..."` or `"0x..."` are still broken as well.

Sorry for putting the work on you. I should get around to cleaning up this code later otherwise.

Could do `value.isalnum()` instead by the way.

Reply:
@galak @ulfalizer Sorry for the long delay in getting back to this PR.

> This parsing code needs a lot of cleanup in general I think. It doesn't make sense to re-enter parse_value() for the contents of a string.

`parse_value` gets re-entered only when `value` is a list. For instance:

value of `<0x02 0x03 0x20000>` will re-enter `parse_value` as the contents of the list as `0x02`, `0x03`, and `0x20000`. And for me, it totally looks fine as the function only parses single value at am instance.

> Could do value.isalnum() instead by the way.

`isalnum` will return false if the string has spaces in it. So we need to ignore the spaces somehow. How about below instead:

```
if value.replace(' ','').isalnum():
```